### ISSUE TITLE
Flake8 doesn't use project configuration from .flake8

### ISSUE THREAD

**Issue by asottile** [PyCQA/flake8#477]
In GitLab by @ziima on Feb 15, 2016, 01:21

```
$ python --version
Python 2.7.9
$ pip --version
pip 1.5.6 from /usr/lib/python2.7/dist-packages (python 2.7)
$ python -c 'import setuptools; print(setuptools.__version__)'
5.5.1
$ flake8 --version
2.5.4 (pep8: 1.6.2, pyflakes: 1.0.0, mccabe: 0.4.0) CPython 2.7.9 on Linux
```

Flake8 takes precedence of user config file localed in `.config/flake8` over project `.flake8` file in working directory. I'd expect that project file will have precedence as described in changelog for 2.5.1.

**Comment by asottile** [comment]
In GitLab by @sigmavirus24 on Feb 15, 2016, 07:39

Try upgrading pep8. Flake8 does not support any pep8 release in the 1.6.x release as this introduced behaviour similar to this.

**Comment by asottile** [comment]
In GitLab by @ziima on Feb 15, 2016, 08:37

OK, that was it.

**Comment by asottile** [comment]
In GitLab by @ziima on Feb 15, 2016, 08:39

My bad, it didn't help. I have added `--config` to my script.

I still doesn't work, unless explicetely defined.

**Comment by asottile** [comment]
In GitLab by @sigmavirus24 on Feb 15, 2016, 15:09

Can you provide the contents of your user config file, project config file, and the output from `flake8 --verbose`

**Comment by asottile** [comment]
In GitLab by @ziima on Feb 16, 2016, 24:10

```
$ cat ~/.config/flake8 
[flake8]
max-line-length = 120

$ cat .flake8 
[flake8]
max-line-length = 60

$ ~/.local/bin/flake8 -v pyvmd/
local configuration: in /home/ziima/phd/git/pyvmd
cli configuration: /home/ziima/.config/flake8
directory pyvmd
checking pyvmd/__init__.py
checking pyvmd/analysis.py
checking pyvmd/analyzer.py
checking pyvmd/atoms.py
checking pyvmd/collectors.py
checking pyvmd/datasets.py
checking pyvmd/labels.py
checking pyvmd/measure.py
checking pyvmd/molecules.py
checking pyvmd/representations.py
checking pyvmd/utils.py
directory pyvmd/tests
checking pyvmd/tests/__init__.py
checking pyvmd/tests/test_analysis.py
checking pyvmd/tests/test_analyzer.py
checking pyvmd/tests/test_atoms.py
checking pyvmd/tests/test_collectors.py
checking pyvmd/tests/test_coloring_methods.py
checking pyvmd/tests/test_datasets.py
checking pyvmd/tests/test_drawing_methods.py
checking pyvmd/tests/test_labels.py
checking pyvmd/tests/test_measure.py
checking pyvmd/tests/test_molecules.py
checking pyvmd/tests/test_representations.py
checking pyvmd/tests/test_utils.py
checking pyvmd/tests/utils.py
pyvmd/tests/utils.py:78:34: E128 continuation line under-indented for visual indent
pyvmd/tests/utils.py:85:34: E128 continuation line under-indented for visual indent
pyvmd/tests/utils.py:92:34: E128 continuation line under-indented for visual indent
pyvmd/tests/utils.py:101:30: E128 continuation line under-indented for visual indent
pyvmd/tests/utils.py:110:30: E128 continuation line under-indented for visual indent
directory pyvmd/tests/data
```

**Comment by asottile** [comment]
In GitLab by @sigmavirus24 on Feb 17, 2016, 06:42

This absolutely is a bug in the way we're constructing `StyleGuide`'s with pep8 because of an annoying catch in how they use the `config_file` parameter to their `StyleGuide`. I have a fix locally that I'm toying with and will be pushing up for code review later.

Thanks for bearing with me and helping me figure this out @ziima

**Comment by asottile** [comment]
In GitLab by @sigmavirus24 on Feb 17, 2016, 18:57

mentioned in commit 533d222f525d0d8c0fecf392d6ec7d211af1d897

**Comment by asottile** [comment]
In GitLab by @sigmavirus24 on Feb 17, 2016, 18:58

mentioned in merge request !53

**Comment by asottile** [comment]
In GitLab by @srisankethu on Mar 28, 2016, 01:20

@sigmavirus24 I doubt if this is the right place  to ask this but I installed flake8 using pip on Ubuntu and i have no flake8 config file at ~/.config ?

**Comment by asottile** [comment]
In GitLab by @sigmavirus24 on Mar 28, 2016, 08:21

@srisankethu you're correct. This is not the right place. All questions belong on [StackOverflow](https://stackoverflow.com/questions/tagged/flake8).

**Comment by asottile** [comment]
In GitLab by @sigmavirus24 on Mar 28, 2016, 08:22

Status changed to closed by commit 533d222f525d0d8c0fecf392d6ec7d211af1d897
